features/integration/c2c.feature

```diff
@@ -0,0 +1,107 @@
+    And I sign and submit the transaction, saving the txid. If there is an error it is "".
+    And I wait for the transaction to be confirmed.
+    Given I remember the new application ID.
+    # Need to fund SlotMachine because it pays for calling RandomByte:
+    And I fund the current application's address with 100000000 microalgos.
+
+    Given a new AtomicTransactionComposer
+
+    # First spin() -> (result, witness0, witness1, witness2)
+    When I create the Method object from method signature "spin(application,application)(byte[3],byte[17],byte[17],byte[17])"
+    * I create a new method arguments array.
+    * I append the encoded arguments "ctxAppIdx:0,ctxAppIdx:1" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    # Second spin() -> (result, witness0, witness1, witness2)
+    When I create the Method object from method signature "spin(application,application)(byte[3],byte[17],byte[17],byte[17])"
+    * I create a new method arguments array.
+    * I append the encoded arguments "ctxAppIdx:0,ctxAppIdx:1" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    # Atomic Transaction Execution and Analysis
+    Then I build the transaction group with the composer. If there is an error it is "".
+    And I gather signatures with the composer.
+    And I execute the current transaction group with the composer.
+    And The composer should have a status of "COMMITTED".
+    And The app should have returned ABI types "(byte[3],byte[17],byte[17],byte[17]):(byte[3],byte[17],byte[17],byte[17])".
+    And I can dig into the resulting atomic transaction execution tree with path "0,0,0"
+    And I can dig into the resulting atomic transaction execution tree with path "0,2,0"
+    And I can dig into the resulting atomic transaction execution tree with path "1,2,0"
+    And I dig into the paths "0,0:0,1:0,2" of the resulting atomic transaction tree I see group ids and they are all the same
+    And I can retrieve all inner transactions that were called from the atomic transaction with call graph "[{'spin(application,application)(byte[3],byte[17],byte[17],byte[17])':[{'appl':'appl'},{'appl':'appl'},{'appl':'appl'}]},{'spin(application,application)(byte[3],byte[17],byte[17],byte[17])':[{'appl':'appl'},{'appl':'appl'},{'appl':'appl'}]}]".
+    And The 0th atomic result for "spin()" satisfies the regex "^[@!-][@!-][@!-]$"
+    And The 1th atomic result for "spin()" satisfies the regex "^[@!-][@!-][@!-]$"
+    And I can dig the 0th atomic result with path "inner-txns.0.inner-txns.0.global-state-delta.0.key" and see the eval delta field "Y291bnRlcg=="
```

Comment:
I'm not sure how we would unify. Maybe something like:
```
... with the path "innerTxns,0,innerTxns,0,globalStateDelta,0,key"...
```
?

Reply:
You could also define a file format for checking all results at once to the expected output file, maybe it would need some sort of placeholders for non-deterministic values?

Reply:
Will take feedback from others who are implementing this step to see if there's a better version